### ISSUE TITLE
improve the error messages of nix-store (e.g. "no operation specified")

### ISSUE THREAD

**Issue by zseri** [NixOS/nix#4892]
**Is your feature request related to a problem? Please describe.**
When calling, e.g. `nix-store --repair /nix/store/...`, nix-store reacts by:
```
error: no operation specified
Try 'nix-store --help' for more information.
```
This error message is a bit unexpected, as an unknown option was specified (which is a misspelled operation).

**Describe the solution you'd like**
It would be nice if this error message could be improved somewhat (ideally printing available operations).

**Describe alternatives you've considered**
none.

**Additional context**

**Comment by alyaeanyx** [comment]
I agree, also the error message for running `nix-store --repair-path` without sudo is quite confusing:
```
$ nix store repair /nix/store/zrzykkzd1s3yxyfw64kivdqffhv6l0x0-openconnect-8.10
error: operation 'repairPath' is not supported by store 'daemon'
```

**Comment by n0099** [comment]
https://github.com/NixOS/nix/issues/11148#issuecomment-2244936414
> * Suggest `sudo` in
> ```
> nix store repair --all 
> error: operation 'repairPath' is not supported by store 'daemon'`
> ```
> * `cannot repair path` -> say it wasn't substitutable* [ ]  Remember corrupted path state in store db

Appeared in:
https://www.raphael.li/blog/2024/06/nix-no-such-file-or-directory-drv/
https://discourse.nixos.org/t/weird-error-om-applying-configuration/59699/3
https://old.reddit.com/r/NixOS/comments/sswctb/how_to_forcerebuild_an_installed_package/

and the old command will hint about permission:
```
nix-store --verify --repair                   
error: you are not privileged to repair paths
```
